app/javascript/controllers/theme_controller.js

```diff
@@ -0,0 +1,65 @@
+import { Controller } from "@hotwired/stimulus"
+
+const THEME_DARK = "dark"
+const THEME_LIGHT = "light"
+
+export default class extends Controller {
+  static targets = ["label"]
+  static values = { storageKey: { type: String, default: "theme-preference" } }
+
+  connect() {
+    this.applyTheme()
+    this.updateAllLabels()
+    document.addEventListener("turbo:frame-load", () => this.updateAllLabels())
+    document.addEventListener("turbo:load", () => this.updateAllLabels())
```

Comment:
I think you could fix the flashing by calling `applyTheme()` in your `turbo:load` event handler.